src/java.base/share/classes/java/lang/Readable.java

```diff
@@ -40,14 +40,17 @@ public interface Readable {
      * Attempts to read characters into the specified character buffer.
      * The buffer is used as a repository of characters as-is: the only
      * changes made are the results of a put operation. No flipping or
-     * rewinding of the buffer is performed.
+     * rewinding of the buffer is performed. If the specified character
+     * buffer is full, then no characters will be read and zero will be
+     * returned.
      *
      * @param cb the buffer to read characters into
      * @return The number of {@code char} values added to the buffer,
-     *                 or -1 if this source of characters is at its end
+     *         possibly zero, or -1 if this source of characters is at its end
```

Comment:
Would it be clearer to specifically indicate 0 is returned when no characters will be read.

Reply:
Might be. I was mimicking the return specification from [FileChannel::read](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/nio/channels/FileChannel.html#read(java.nio.ByteBuffer[],int,int)).

Reply:
Consistency is probably good I guess in this case, though specifying `zero` vs `0` seems inconsistent when we specify `-1` as I believe this varies throughout the javadoc.  But it matches the other javadoc so we are at least consistent (somewhat )

OK for now by me